adr/ADR-43.md

```diff
@@ -52,21 +49,40 @@ The `Nats-Limit-Applied` field is there to support future expansion of this feat
 
 This behaviour is off by default unless opted in on the Stream Configuration.
 
+### Sources and Mirrors
+
+Sources and Mirrors will always accept and store messages with `Nats-TTL` header present, even if the `AllowMsgTTL` setting is disabled in the Stream settings.
+
+If the `AllowMsgTTL` setting is enabled then processing continues as outlined in the General Behavior section with messages removed after the TTL. With the setting disabled the messages are just stored.
+
+Sources may set the `SubjectDeleteMarkers` option and processing of messages with the `Nats-TTL` will place tombstones, but, Mirrors may not enable `SubjectDeleteMarkers` since it would insert new messages into the Stream it might make it impossible to match sequences from the Mirrored Stream.
+
 ## Stream Configuration
 
-Weather or not a stream support this behavior should be a configuration opt-in. We want clients to definitely know 
-when this is supported which the opt-in approach with a boolean on the configuration would make clear.
+Weather or not a stream support this behavior should be a configuration opt-in. We want clients to definitely know when this is supported which the opt-in approach with a boolean on the configuration would make clear.
 
-We have to assume someone will want to create a replication topology where at some point in the topology these tombstone
-type messages are retained for an audit trail. So a Stream with this feature enabled can replicate to one with it 
-disabled and all the messages that would have been TTLed will be retained.
+We have to assume someone will want to create a replication topology where at some point in the topology these tombstone type messages are retained for an audit trail. So a Stream with this feature enabled can replicate to one with it disabled and all the messages that would have been TTLed will be retained.
 
 ```golang
 type StreamConfig struct {
 	// AllowMsgTTL allows header initiated per-message TTLs
 	AllowMsgTTL bool          `json:"allow_msg_ttl"`
 
-	// LimitsTTL activates writing of messages when limits are applied with a specific TTL
-	LimitsTTL   time.Duration `json:"limits_ttl"`
+	// Enables placing markers in the stream for certain message delete operations
+	SubjectDeleteMarkers   bool   `json:"subject_delete_markers,omitempty"`
```

Comment:
This setting is really not needed - the TTL setting is going to be highly dependent on the use-case, so auto-setting it is not the right approach - user should just specify the `SubjectDeleteMarkerTTL` and that infers that `SubjectDeleteMarkers`

Reply:
We had this conversation quite a few times, but the prevailing problem is that unless KVs etc are setting it for you, that users setting this on their own streams may have no way to intuitively know what a sensible value even is, i.e. for lost connections, ordered consumer recreation, etc, hence the "sensible default".

Reply:
We also do not know what a correct setting is.  If a user cannot pick a default, neither can we.  All we are doing is guaranteeing that people will feel the system is broken rather than that it is behaving in a way that they chose - even if at the time of choosing they might not have realised the full weight of the choice.

We are still pretending we can answer the question: What is the default outage time? 

We can not.

Further this pushes a bunch of complexity onto everywhere, want to unset this setting now there are other coupled settings to also unset and understand etc.

I am, still, strongly against this design.